### ISSUE TITLE
Support !php/const in routing yaml

### ISSUE THREAD

**Issue by timwsuqld** [symfony/symfony#31356]
**Description**  
Currently we support `!php/const` for constants in service files. It would be good to be able to use them in routing files as well.

**Example**  
We have some routes that rely on constants, we currently have to put them in to a parameter and then we can use them in our routes

Currently we have:
```yaml
staff_dashboard:
    host: '%domain_staff%'
    path: /dashboards/default/
```
Plus a parameter for domain_staff which actually comes from a constant `DOMAIN_STAFF`.

We could do away with the parameter and just have:
```yaml
staff_dashboard:
    host: !php/const DOMAIN_STAFF
    path: /dashboards/default/
```

**Fix**
In YamlFileLoader.php we change
```php
$parsedConfig = $this->yamlParser->parseFile($path);
```
to
```php
$parsedConfig = $this->yamlParser->parseFile($path, Yaml::PARSE_CONSTANT);
```

**Comment by xabbuh** [comment]
That's already possible because of the changes made in #25293 which was first released in 4.1.0.

**Comment by timwsuqld** [comment]
@xabbuh Thanks for that. I did search for bugs but didn't find it. I'm in the process of moving to flex in 3.4, so we can upgrade to Symfony 4, so it's good to know once I get there we can make those changes.

Sorry for the noise
